### ISSUE TITLE
[FEATURE]: make Node max heap size configurable

### ISSUE THREAD

**Issue by vvagaytsev** [garden-io/garden#5320]
### Background / Motivation

Currently, Garden uses a hard-coded max heap size for Node, see https://github.com/garden-io/garden/blob/main/bin/garden#L5:
```bash
node --max-old-space-size=4096 --max-semi-space-size=64 ${repo_root}/cli/bin/garden "$@"
```

Sometimes this causes OOMs.  The recently reported OOMs appear not to be memory leaks as people not seeing memory always increasing.

### What should the user be able to do?

User should be able to override the max heap size if necessary.

### Why do they want to do this? What problem does it solve?

This will solve OOMs for heavy projects.

### Suggested Implementation(s)

Introduce an env variable to control the heap size? something like `GARDEN_MAX_HEAP_SIZE_MB`.

### How important is this feature for you/your team?

🌵 Not having this feature makes using Garden painful

**Comment by stefreak** [comment]
With our current way of building the garden single executable binary, vercel/pkg, this is not possible, as the node flags are hardcoded in the binary.

In the ESM PR (#5233) we currently work on moving away from PKG to our own "single executable" solution written in rust (https://github.com/garden-io/garden/tree/chore/esm/garden-sea) – that will give us the control needed to implement controlling the heap size at runtime.

**Comment by stefreak** [comment]
This has been implemented in the ESM PR (#5233) 

These settings can now be overridden using the environment variables `GARDEN_MAX_OLD_SPACE_SIZE` and `GARDEN_MAX_SEMI_SPACE_SIZE`
